node/src/blockchain/blockchain_interface_utils.rs

```diff
@@ -0,0 +1,1412 @@
+
+        let _result = sign_transaction(
+            chain,
+            Web3::new(Batch::new(transport)),
+            recipient_wallet,
+            consuming_wallet,
+            amount,
+            nonce,
+            gas_price_in_gwei,
+        );
+    }
+
+    #[test]
+    #[should_panic(expected = "Signing should be done locally")]
+    fn sign_transaction_locally_panics_on_signed_transaction() {
+        let port = find_free_port();
+        let (_event_loop_handle, transport) = Http::with_max_parallel(
+            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
+            REQUESTS_IN_PARALLEL,
+        )
+        .unwrap();
+        let chain = DEFAULT_CHAIN;
+        let amount = 11_222_333_444;
+        let gas_limit = U256::from(5);
+        let gas_price = U256::from(5);
+        let recipient_wallet = make_wallet("recipient_wallet");
+        let consuming_wallet = make_paying_wallet(b"consuming_wallet");
+        let data = sign_transaction_data(amount, recipient_wallet);
+        // sign_transaction makes a blockchain call because nonce is set to None
+        let transaction_parameters = TransactionParameters {
+            nonce: None,
+            to: Some(chain.rec().contract),
+            gas: gas_limit,
+            gas_price: Some(gas_price),
+            value: U256::zero(),
+            data: Bytes(data.to_vec()),
+            chain_id: Some(chain.rec().num_chain_id),
+        };
+        let key = consuming_wallet
+            .prepare_secp256k1_secret()
+            .expect("Consuming wallet doesnt contain a secret key");
+
+        let _result = sign_transaction_locally(
+            Web3::new(Batch::new(transport)),
+            transaction_parameters,
+            &key,
+        );
+    }
+
+    #[test]
+    fn sign_and_append_payment_just_works() {
+        let port = find_free_port();
+        let (_event_loop_handle, transport) = Http::with_max_parallel(
+            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
+            REQUESTS_IN_PARALLEL,
+        )
+        .unwrap();
+        let consuming_wallet = make_paying_wallet(b"consuming_wallet");
+        let system = System::new("test");
+        let account = make_payable_account_with_wallet_and_balance_and_timestamp_opt(
+            make_wallet("blah123"),
+            9000,
+            None,
+        );
+        let gas_price = 123;
+        let nonce = U256::from(1);
+
+        let result = sign_and_append_payment(
+            TEST_DEFAULT_CHAIN,
+            Web3::new(Batch::new(transport)),
+            consuming_wallet,
+            nonce,
+            gas_price,
+            account,
+        );
+
+        System::current().stop();
+        system.run();
+        let expected_hash =
+            H256::from_str("8d278f82f42ee4f3b9eef2e099cccc91ff117e80c28d6369fec38ec50f5bd2c2")
+                .unwrap();
+        assert_eq!(result.hash, expected_hash);
+        assert_eq!(result.amount, 9000);
+    }
+
```

Comment:
I do believe, but Utkarsh's opinion is welcomed, that all the suit of tests below this line are actually an example of huge misunderstanding. We've dragged them along for years, even when Substratum was thriving, but the truth about them is that it has nothing to do with our code and that they test purely and alone code of the web3 library. This efforts is valueless in my eyes. And I'm all for getting rid of the garbage. Except a few tests that do belong into our codebase and those are these three:

```
    #[test]
    fn gas_limit_for_polygon_mumbai_lies_within_limits_for_raw_transaction() {
        test_gas_limit_is_between_limits(Chain::PolyAmoy);
    }

    #[test]
    fn gas_limit_for_polygon_mainnet_lies_within_limits_for_raw_transaction() {
        test_gas_limit_is_between_limits(Chain::PolyMainnet);
    }

    #[test]
    fn gas_limit_for_eth_mainnet_lies_within_limits_for_raw_transaction() {
        test_gas_limit_is_between_limits(Chain::EthMainnet)
    }
```